core/amber/src/main/python/core/architecture/sendsemantics/hash_based_shuffle_partitioner.py

```diff
@@ -42,6 +42,15 @@ def add_tuple_to_batch(
             yield receiver, batch
             self.receivers[hash_code] = (receiver, list())
 
+    @overrides
+    def add_state_to_batch(self, state: State):
+        for receiver, batch in self.receivers:
+            if len(batch) > 0:
+                yield receiver, deepcopy(batch)
+            yield receiver, deepcopy(batch)
```

Comment:
Two issues:
1. can we avoid deepcopies? 
2. why are you yielding out the same batch twice?

Reply:
function removed.